packages/react-meteor-data/suspense/useSubscribe.ts

```diff
@@ -0,0 +1,63 @@
+import { useEffect } from 'react'
+import { EJSON } from 'meteor/ejson'
+import { Meteor } from 'meteor/meteor'
+import isEqual from 'lodash/isEqual'
+import { removeFromArray } from './useFind'
+
+const cachedSubscriptions: SubscriptionEntry[] = []
+
+interface SubscriptionEntry {
+  params: EJSON[]
+  name: string
+  handle?: Meteor.SubscriptionHandle
+  promise: Promise<Meteor.SubscriptionHandle>
+  result?: Meteor.SubscriptionHandle
+  error?: unknown
+}
+
+export function useSubscribeSuspense(name: string, ...params: EJSON[]) {
+  const cachedSubscription =
+    cachedSubscriptions.find(x => x.name === name && isEqual(x.params, params))
+
+  useEffect(() => {
+    return () => {
+      setTimeout(() => {
+        if (cachedSubscription != null) {
+          cachedSubscription?.handle?.stop()
+          removeFromArray(cachedSubscriptions, cachedSubscription)
+        }
+      }, 0)
+    }
+  }, [name, ...params])
+
+  if (cachedSubscription != null) {
+    if ('error' in cachedSubscription) throw cachedSubscription.error
+    if ('result' in cachedSubscription) return cachedSubscription.result
+    throw cachedSubscription.promise
+  }
+
+  const subscription: SubscriptionEntry = {
+    name,
+    params,
+    promise: new Promise<Meteor.SubscriptionHandle>((resolve, reject) => {
+      const h = Meteor.subscribe(name, ...params, {
+        onReady() {
+          subscription.result = h
```

Comment:
I'm just thinking whether the handle is actually useful here - it's always going to be ready, and it doesn't make sense to `stop` it. Maybe we should not return it in the first version and see if it's needed afterwards?

Reply:
I think the same. I only need it to stop when the component unmounts. I can hide it from the end user with the `.handle` and return just null for the end user